pkg/types/validation/installconfig.go

```diff
@@ -300,7 +313,8 @@ func validateClusterNetwork(n *types.Networking, cn *types.ClusterNetworkEntry,
 	if cn.HostPrefix < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("hostPrefix"), cn.HostPrefix, "hostPrefix must be positive"))
 	}
-	if ones, bits := cn.CIDR.Mask.Size(); cn.HostPrefix < int32(ones) {
+	// ignore hostPrefix if the plugin does not use it and has it unset
+	if ones, bits := cn.CIDR.Mask.Size(); (networkTypeRequiresHostPrefix(n.NetworkType) || (cn.HostPrefix != 0)) && cn.HostPrefix < int32(ones) {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("hostPrefix"), cn.HostPrefix, "cluster network host subnetwork prefix must not be larger size than CIDR "+cn.CIDR.String()))
 	} else if bits == 128 && cn.HostPrefix != 64 {
```

Comment:
This will get checked even if `networkTypeRequiresHostPrefix` returned `false`. You want to split the `networkTypeRequiresHostPrefix || != 0` check out to a separate `if` wrapping this section

Reply:
uh-oh you are right, I guess this got added on my last rebase. sorry for not having realised this.